log4j-api/src/main/java/org/apache/logging/log4j/util/ServiceLoaderUtil.java

```diff
@@ -47,10 +57,20 @@ private ServiceLoaderUtil() {
      * 
      * @param <T>         The service type.
      * @param serviceType The class of the service.
-     * @return List of available services.
+     * @return Stream of available services.
      */
-    public static <T> List<T> loadServices(final Class<T> serviceType) {
-        return loadServices(serviceType, null);
+    public static <T> Stream<T> loadServices(final Class<T> serviceType) {
```

Comment:
This is an api/abi break moving from `List<T>` to `Stream<T>`. The Stream allows us to avoid unused service-loader lookups, but should be added as a new method where we deprecate the original.

Reply:
I added the `ServiceLoaderUtil` class after the 2.17.2 release, so there is no API break for now.